pandas/core/indexes/multi.py

```diff
@@ -1728,7 +1728,7 @@ def get_level_values(self, level):
     def unique(self, level=None):
 
         if level is None:
-            return super().unique()
```

Comment:
Looks like this old path would trigger a `_shallow_copy` and it doesn't look like `drop_duplicates` does. Any implications off the top of your head? (Maybe another reason why there's a perf improvement)

Reply:
Hm not really, we are returning a view on the original data anyway if the MultiIndex is unique, so the duplicate case does not have to trigger a _shallow_copy I think.

The performance improvement has another cause:

Before this, we cast our values to a numpy array of tuples. This pr avoids that.